demos/dashy/src/index.ts

```diff
@@ -24,52 +25,51 @@ const origSend: SendFunc = hookSend(function mySend(opts: IOptions, action: stri
 */
 
 export class App {
-    _dashy = new Dashy();
+    _dashy: Dashy | Dashboard;
     _layoutJson = {};
 
     constructor(placeholder: string) {
-        this._dashy
-            .target(placeholder)
-            .render()
-            ;
-
-        this._dashy.element()
-            .on("drop", () => this.dropHandler(this.event()))
-            .on("dragover", () => this.dragOverHandler(this.event()))
-            ;
-        if (document.URL.indexOf("dsp=") !== -1) {
-            const dsp = document.URL.split("dsp=")[1].split("&")[0].split("%26").join("&");
-            fetch(dsp)
-                .then(resp => resp.json())
-                .then(json => {
-                    const protocol = json.response.LayoutText.__properties.ddlUrl.split("://")[0];
-                    const hostname = json.response.LayoutText.__properties.ddlUrl.split(":")[1].slice(2);
-                    const port = json.response.LayoutText.__properties.ddlUrl.split(":")[2].split("/")[0];
-                    const urlStr = json.response.LayoutText.__properties.ddlUrl + `&Protocol=${protocol}&Hostname=${hostname}&Port=${port}`;
-
-                    this.parseUrl(urlStr, json.response.LayoutText);
-                })
-                ;
-        } else {
-            this.parseUrl(document.URL);
-        }
+        this.parseUrl(placeholder, document.URL);
     }
 
     event() {
         return d3Event || event;
     }
 
-    parseUrl(urlStr, layoutJson?) {
+    importDDL(ddlStr: string, baseUrl: string, wuid: string, layoutJson) {
+        const ddl = JSON.parse(ddlStr);
+        const ddl2: DDL2.Schema = isDDL2Schema(ddl) ? ddl : upgrade(ddl, baseUrl, wuid, true, layoutJson);
+        this._dashy.restore(ddl2);
+    }
+
+    parseUrl(placeholder: string, urlStr, layoutJson?) {
```

Comment:
This method name seems to be no longer relevant since a lot more than parsing the URL is taking place.